### ISSUE TITLE
🐛[bug] already  set  min_validation_period , but still got a single validation metric

### ISSUE THREAD

**Issue by scotthuang1989** [determined-ai/determined#9863]
### Describe the bug

here is my training configuration:

records_per_epoch: 1000000
min_validation_period:
  batches: 1000

the training has run for 10430 batches. but validation only run 1 times at the end of training.

### Reproduction Steps

1.
2.
3.

### Expected Behavior

validation run every 1000 batches. 

### Screenshot

![image](https://github.com/user-attachments/assets/0eb6901e-e061-459b-9146-6b0e913d3fab)

### Environment

- Device or hardware: [e.g. iPhone6, Nvidia A100]
- Version [0.35.0]

**Comment by ioga** [comment]
hello, 

any chance you can provide a repro code? that'd be an ideal way to track this down. 

otherwise, we may only guess, and we'll need more details for that:
1. which trial or trainer class are you using (PyTorchTrial, TFKerasTrial, torch trainer, Core API, etc.) 
2. dataloader details

**Comment by scotthuang1989** [comment]
sorry for the delay.  I am unable to  reproduce this on a public dataset. I attach my training log, hope it helps. 
and I will continue to reproduce it, thanks.
[experiment_83_trial_386_logs.txt](https://github.com/user-attachments/files/16781752/experiment_83_trial_386_logs.txt)

**Comment by ioga** [comment]
that log is not particularly helpful in answering the questions. with some dataloader setups, the actual dataset size matters for the behavior. 

the log has a weird message though: `[2024-08-28T02:21:48.687318Z] e2563a43 || val dataset is None, use train data`

**Comment by scotthuang1989** [comment]
Hi ioga, 

I found the cause of this issue.
I use following code to start the experiments

```    
with pytorch.init(hparams=hparams) as train_context:
        trial = PredictionTrial(train_context, hparams=hparams)
        trainer = pytorch.Trainer(trial, train_context)
        trainer.fit(max_length=max_length, latest_checkpoint=latest_checkpoint)
```

I need pass the min_checkpoint_period to "trainer.fit" manually, determinedai will not do for me.

thanks.
